### DISCUSSION TITLE
How to create and run automated tests for WordPress blocks

### DISCUSSION THREAD

**Discussion by juanmaguitar** [WordPress/developer-blog-content#24] [Topic Ideas]
An article about how to create and run automated tests for our custom blocks in WordPress could be useful for both the product itself and the extenders using blocks. 

In fact, I see this content as a series of two posts:
- **Unit Tests for Blocks**: The first one covers a more granular way of testing blocks: Unit Tests with Jest to test a block from its main components (`Edit` and `save`)
- **E2E for Blocks**:  A second one covering a more generic way of testing blocks: End-to-end tests with playwright to test the expected block behavior in the WordPress context (the block is properly registered, the frontend behavior works as expected, ...)

The first post could be called "Unit Testing for Blocks" and could cover the following ideas:
- Why having automated tests is a good way to ensure our block work as expected and to document it (among other benefits)
- The difference between Unit Tests and E2E tests
- What is Jest 
- Preparing the environment in our project to test our block with Jest
- What to test in a Unit Test for a block - creating some tests for our block
- Launching our test in local
- Bonus: Creating a CI/CD pipeline with GitHub Actions to only allow merge if tests are passing

These posts could even include a brief explanation of what [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development) is, what [CI](https://en.wikipedia.org/wiki/Continuous_integration) is, and how these tests allow the Gutenberg project to grow in a more solid way.

Inspired by [this answer](https://wordpress.org/support/topic/the-navigation-toggle-menu-button-does-not-work-on-localhost/#post-16239250) in the support forums

**Comment by juanmaguitar** [comment]
@bph, is there anything I can do to re-surface this Topic Idea?
I'd be happy to write an article about this in Q3

**Reply by bph** [reply]
It would help if you elaborate on specifically what your article might entail. The current information a bit too broad or too short. Although theory is always great, writing how to implement for specific use cases has been a great approach, readers on the developer blog, appreciate.

**Reply by bph** [reply]
If I recall it correctly, when this topic came up, the Gutenberg contributors were working on adding playwright tests to the project. Now that this migration is done it would be wonderful to have a how to create a test using Playwright.

**Reply by juanmaguitar** [reply]
Thanks for the guidance!
I'll revamp the description to include more details about what I would like to include in this article.

**Comment by bph** [comment]
This topic has been approved. Please continue the conversation on the issue

- #269

**Comment by meszarosrob** [comment]
Published at https://developer.wordpress.org/news/2026/05/getting-started-writing-wordpress-e2e-tests-with-playwright/
